nsot/models.py

```diff
@@ -858,115 +859,83 @@ def get_next_network(self, prefix_length, num=None, as_objects=True):
 
         cidr = self.ip_network
 
-        # If this network is an interconnect network, generate the hosts as
-        # network objects, otherwise just subnet based on the prefix_length.
-        # This is so that .0 and .1 could be allocated from a /31, for example.
-        if cidr.prefixlen in settings.NETWORK_INTERCONNECT_PREFIXES:
-            log.debug('CIDR %s is an interconnect!', cidr)
-            subnets = (ipaddress.ip_network(ip) for ip in cidr)
-        else:
-            log.debug('cidr %s is NOT an interconnect!', cidr)
-            subnets = cidr.subnets(new_prefix=prefix_length)
-
-        # Exclude children that are in busy states.
-        children = self.get_descendants()
-
-        # Partition children into busy children and all children, storing
-        # the `.ip_network` property up front.
-        busy_children = [
-            c.ip_network for c in children if (
-                c.state in self.BUSY_STATES
-            )
-        ]
-        children = [
-            c.ip_network for c in children
-        ]
-
-        # Prepopulate children_seen with networks that have the same prefix
-        # length as wanted `prefix_length`.
-        children_seen = [
-            c for c in children if (
-                c.prefixlen == prefix_length
-            )
-        ]
-
-        # Dirty subnets cannot be used. Pre-seed w/ busy children.
-        dirty = [c for c in busy_children if (c.prefixlen == prefix_length)]
-
-        # Pre-filter any pre-seeded dirty subnets as a generator.
-        subnets = (s for s in subnets if s not in dirty)
-
-        # Subnets we want.
-        wanted = []
-
-        # Keep iterating until we've found the number of networks we want.
-        while len(wanted) < num:
+        if prefix_length > cidr.max_prefixlen:
             try:
-                next_subnet = next(subnets)
+                next(cidr.subnets(new_prefix = prefix_length))
             except ValueError as err:
                 raise exc.ValidationError({'prefix_length': err.message})
-            except StopIteration:
-                break
 
-            log.debug('>> CHILDREN_SEEN: %s items, DIRTY: %s items',
-                      len(children_seen), len(dirty))
+        if strict:
+            children = [c.ip_network for c in self.get_children()]
+        else:
+            children = [c.ip_network for c in self.get_descendants() 
+                            if c.prefix_length >= prefix_length]
+
+        exclude_nums = {}
+        
+        network_prefix = cidr.network_address 
+        # get all numbers formed by bits in child network address
+        # that do not exist in parent network address
+        a = int(network_prefix) >> (cidr.max_prefixlen - prefix_length)
+        for c in children:
```

Comment:
This is great!

One last nitpick: Can you please add some extra commenting to explain what is going on here, so that when this is no longer fresh on our minds we can remember

Thanks in advance!